### PR TITLE
Sy 945 server relay deadlock on sudden stream transport disconnect

### PR DESCRIPTION
**Pull request by emilbon99**
# Fix Pull Request Template

## Key Information

- **Linear Issue**: [SY-945](https://linear.app/synnaxlabs/issue/SY-945/[server]-relay-deadlock-on-sudden-stream-transport-disconnect)

## Description

Adds a write deadline to the server side freighter web socket stream. This means that slow telemetry consumers will eventually be kicked off of the relay. The deadline is configurable with the option `slow-consumer-timeout`.
 
Also renames various `EncoderDecoder` elements to `codec`.

## Basic Readiness Checklist

- [x] I have performed a self-review of my code.
- [x] I have added sufficient regression tests to cover the changes to CI.
- [x] I have added relevant tests to cover the changes or exposing bugs.
- [x] I have verified code coverage targets are met.

## Additional Notes

- [] These changes deal with concurrency.
- [] These changes affect UI.

## Manual QA Additions

- [x] I have updated the [Release Candidate](/.github/PULL_REQUEST_TEMPLATE/rc.md) template
  with necessary manual QA steps to test my change.

## Reviewer Checklist

- [ ] Sufficient test coverage of new additions.
- [ ] Verified all steps in readiness checklists.
- [ ] UI changes have been tested.
- [x] style and formatting is consistent.
- [ ] Reviewed any relevant changes to concurrent code for safety. 
- [ ] Sufficient comments and clarity of code.

### CODE REVIEW

**Review by Lham42 — APPROVED** [review]

**Comment by Lham42** [comment]
LGTM

File: freighter/go/fhttp/stream.go [hunk 1]
```diff
@@ -43,66 +43,81 @@ const (
 	closeMessage    messageType = "close"
```
Comment (LeonLiur) [inline freighter/go/fhttp/stream.go:43]: Can we rename `messageTypeData` to just `data`? similarly `closeMessage` to something else? (I know `close` collids with a builtin, but at least the two variable name formats should be consistent)

File: freighter/go/fhttp/stream.go [hunk 2]
```diff
@@ -152,15 +167,15 @@ func (s *streamClient[RQ, RS]) Stream(
 			if res.StatusCode != fiber.StatusSwitchingProtocols {
 				return oCtx, roacherrors.New("[ws] - unable to upgrade connection")
 			}
-			stream = &clientStream[RQ, RS]{core: newCore[RS, RQ](ctx, conn, s.ecd, s.logger)}
+			stream = &clientStream[RQ, RS]{streamCore: newStreamCore[RS, RQ](ctx, coreConfig{conn: conn, codec: s.codec, Instrumentation: s.Instrumentation})}
```
Comment (LeonLiur) [inline freighter/go/fhttp/stream.go:170]: Format and make multi-line

File: synnax/cmd/start.go [hunk 3]
```diff
@@ -235,16 +236,19 @@ func start(cmd *cobra.Command) {
 		}
 
 		// Configure the HTTP API Transport.
-		r := fhttp.NewRouter(fhttp.RouterConfig{Instrumentation: ins})
+		r := fhttp.NewRouter(fhttp.RouterConfig{
+			Instrumentation:     ins,
+			StreamWriteDeadline: viper.GetDuration("slow-consumer-timeout"),
```
Comment (LeonLiur) [inline synnax/cmd/start.go:241]: Should this use the variable for the flag?

**Review by LeonLiur — APPROVED** [review]

File: x/go/confluence/source.go [hunk 4]
```diff
@@ -44,17 +44,23 @@ func (ams *AbstractMultiSource[V]) SendToEach(ctx context.Context, v V) error {
 func (ams *AbstractMultiSource[V]) SendToEachWithTimeout(
```
Comment (LeonLiur) [inline x/go/confluence/source.go:44]: Could you document this function and each argument?

**Review by LeonLiur — APPROVED** [review]
Approved after you address my two comments. Also, could you briefly explain how this fixes teh original problem of "relay deadlock on stream sudden transport disconnect"? Was the problem that without the timeout, the server cannot close because a listener is always trying to receive even though it's closed?

**Comment by emilbon99** [comment]
The problem was that very slow streamers (essentially blocked) in Python and TS would back up the telemetry relay, preventing it from efficiently sending samples, and slowing the whole system down. This PR introduces explicit mechanisms for severing the stream transport connection if it gets blocked.

File: freighter/go/fhttp/stream.go (hunk 1 above)
Comment (emilbon99) [reply freighter/go/fhttp/stream.go:43]: Yep

File: freighter/go/fhttp/stream.go (hunk 2 above)
Comment (emilbon99) [reply freighter/go/fhttp/stream.go]: OK

**Pull request merged**